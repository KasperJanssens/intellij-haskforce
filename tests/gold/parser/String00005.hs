module String00005 where

instance (Show e) => Show (UntypedFeldF e) where
   show (Variable v)                = show v
   show (Lambda v e)                = "(\\" ++ show v ++ " -> " ++ show e ++ ")"
