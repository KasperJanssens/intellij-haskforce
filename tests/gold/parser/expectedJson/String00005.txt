[{"Module":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":6,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":1}]},{"ModuleHead":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":25,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":20}]},{"ModuleName":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":19,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":8},"srcInfoPoints":[]},"String00005"]},null,null]},[],[],[{"InstDecl":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":3,"srcSpanEndLine":6,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":9,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":49,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":44},{"srcSpanFilename":"A.hs","srcSpanEndColumn":4,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":4},{"srcSpanFilename":"A.hs","srcSpanEndColumn":4,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":4},{"srcSpanFilename":"A.hs","srcSpanEndColumn":0,"srcSpanStartLine":6,"srcSpanEndLine":6,"srcSpanStartColumn":1}]},{"CxParen":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":10},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":11,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":10},{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":17},{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":19}]},{"CxSingle":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":11},"srcInfoPoints":[]},{"ClassA":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":11},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":15,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":11},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":15,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":11},"srcInfoPoints":[]},"Show"]}]},[{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":16},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":16},"srcInfoPoints":[]},"e"]}]}]]}]}]},{"IHead":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":26,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":22},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":26,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":22},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":26,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":22},"srcInfoPoints":[]},"Show"]}]},[{"TyParen":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":43,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":27},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":28,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":27},{"srcSpanFilename":"A.hs","srcSpanEndColumn":43,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":42}]},{"TyApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":42,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":28},"srcInfoPoints":[]},{"TyCon":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":40,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":28},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":40,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":28},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":40,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":28},"srcInfoPoints":[]},"UntypedFeldF"]}]}]},{"TyVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":42,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":41},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":42,"srcSpanStartLine":3,"srcSpanEndLine":3,"srcSpanStartColumn":41},"srcInfoPoints":[]},"e"]}]}]}]}]]},[{"InsDecl":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":81,"srcSpanStartLine":4,"srcSpanEndLine":5,"srcSpanStartColumn":4},"srcInfoPoints":[]},{"FunBind":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":81,"srcSpanStartLine":4,"srcSpanEndLine":5,"srcSpanStartColumn":4},"srcInfoPoints":[]},[{"Match":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":45,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":4},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":8,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":4},"srcInfoPoints":[]},"show"]},[{"PParen":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":9},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":10,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":9},{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":20}]},{"PApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":10},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":10},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":10},"srcInfoPoints":[]},"Variable"]}]},[{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":20,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":19},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":20,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":19},"srcInfoPoints":[]},"v"]}]}]]}]}],{"UnGuardedRhs":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":45,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":37},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":38,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":37}]},{"App":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":45,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":39},"srcInfoPoints":[]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":43,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":39},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":43,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":39},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":43,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":39},"srcInfoPoints":[]},"show"]}]}]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":45,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":44},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":45,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":44},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":45,"srcSpanStartLine":4,"srcSpanEndLine":4,"srcSpanStartColumn":44},"srcInfoPoints":[]},"v"]}]}]}]}]},null]},{"Match":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":81,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":4},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":8,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":4},"srcInfoPoints":[]},"show"]},[{"PParen":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":9},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":10,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":9},{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":20}]},{"PApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":16,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":10},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":16,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":10},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":16,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":10},"srcInfoPoints":[]},"Lambda"]}]},[{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":17},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":18,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":17},"srcInfoPoints":[]},"v"]}]},{"PVar":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":20,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":19},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":20,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":19},"srcInfoPoints":[]},"e"]}]}]]}]}],{"UnGuardedRhs":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":81,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":37},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":38,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":37}]},{"InfixApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":81,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":39},"srcInfoPoints":[]},{"Lit":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":44,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":39},"srcInfoPoints":[]},{"String":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":44,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":39},"srcInfoPoints":[]},"(\\","(\\\\"]}]},{"QVarOp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":47,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":45},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":47,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":45},"srcInfoPoints":[]},{"Symbol":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":47,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":45},"srcInfoPoints":[]},"++"]}]}]},{"InfixApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":81,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":48},"srcInfoPoints":[]},{"App":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":54,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":48},"srcInfoPoints":[]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":52,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":48},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":52,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":48},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":52,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":48},"srcInfoPoints":[]},"show"]}]}]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":54,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":53},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":54,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":53},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":54,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":53},"srcInfoPoints":[]},"v"]}]}]}]},{"QVarOp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":57,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":55},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":57,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":55},"srcInfoPoints":[]},{"Symbol":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":57,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":55},"srcInfoPoints":[]},"++"]}]}]},{"InfixApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":81,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":58},"srcInfoPoints":[]},{"Lit":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":64,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":58},"srcInfoPoints":[]},{"String":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":64,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":58},"srcInfoPoints":[]}," -\u003e "," -\u003e "]}]},{"QVarOp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":67,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":65},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":67,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":65},"srcInfoPoints":[]},{"Symbol":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":67,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":65},"srcInfoPoints":[]},"++"]}]}]},{"InfixApp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":81,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":68},"srcInfoPoints":[]},{"App":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":74,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":68},"srcInfoPoints":[]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":72,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":68},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":72,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":68},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":72,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":68},"srcInfoPoints":[]},"show"]}]}]},{"Var":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":74,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":73},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":74,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":73},"srcInfoPoints":[]},{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":74,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":73},"srcInfoPoints":[]},"e"]}]}]}]},{"QVarOp":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":77,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":75},"srcInfoPoints":[]},{"UnQual":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":77,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":75},"srcInfoPoints":[]},{"Symbol":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":77,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":75},"srcInfoPoints":[]},"++"]}]}]},{"Lit":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":81,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":78},"srcInfoPoints":[]},{"String":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":81,"srcSpanStartLine":5,"srcSpanEndLine":5,"srcSpanStartColumn":78},"srcInfoPoints":[]},")",")"]}]}]}]}]}]}]},null]}]]}]}]]}]]},[]]