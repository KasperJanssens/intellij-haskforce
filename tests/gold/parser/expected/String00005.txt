Haskell File
  HaskellModuledeclImpl(MODULEDECL)
    PsiElement(HaskellTokenType.module)('module')
    HaskellQconidImpl(QCONID)
      PsiElement(HaskellTokenType.conidRegexp)('String00005')
    PsiElement(HaskellTokenType.where)('where')
  HaskellBodyImpl(BODY)
    PsiElement(HaskellTokenType.instance)('instance')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.()('(')
    HaskellQconidImpl(QCONID)
      PsiElement(HaskellTokenType.conidRegexp)('Show')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('e')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.,)(',')
    HaskellQconidImpl(QCONID)
      PsiElement(HaskellTokenType.conidRegexp)('Show')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('f')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.))(')')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.=>)('=>')
    HaskellQconidImpl(QCONID)
      PsiElement(HaskellTokenType.conidRegexp)('Show')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.()('(')
    HaskellQconidImpl(QCONID)
      PsiElement(HaskellTokenType.conidRegexp)('UntypedFeldF')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('e')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('f')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.))(')')
    PsiElement(HaskellTokenType.where)('where')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('show')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.()('(')
    HaskellQconidImpl(QCONID)
      PsiElement(HaskellTokenType.conidRegexp)('Variable')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('v')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('t')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.))(')')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.=)('=')
    PsiElement(HaskellTokenType.chartoken)(''m'')
    HaskellSymbolImpl(SYMBOL)
      PsiElement(HaskellTokenType.:)(':')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.()('(')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('show')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('v')
    HaskellQinfixvaridImpl(QINFIXVARID)
      PsiElement(HaskellTokenType.`)('`')
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('append')
      PsiElement(HaskellTokenType.`)('`')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('show')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('t')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.))(')')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('show')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.()('(')
    HaskellQconidImpl(QCONID)
      PsiElement(HaskellTokenType.conidRegexp)('Lambda')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('v')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('e')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.))(')')
    HaskellReservedopImpl(RESERVEDOP)
      PsiElement(HaskellTokenType.=)('=')
    HaskellPstringtokenImpl(PSTRINGTOKEN)
      PsiElement(HaskellTokenType.")('"')
      PsiElement(HaskellTokenType.STRINGTOKEN)('(')
      PsiElement(HaskellTokenType.STRINGTOKEN)('\\')
      PsiElement(HaskellTokenType.")('"')
    HaskellQvarsymImpl(QVARSYM)
      HaskellVarsymImpl(VARSYM)
        HaskellSymbolImpl(SYMBOL)
          PsiElement(HaskellTokenType.+)('+')
        HaskellSymbolImpl(SYMBOL)
          PsiElement(HaskellTokenType.+)('+')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('show')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('v')
    HaskellQvarsymImpl(QVARSYM)
      HaskellVarsymImpl(VARSYM)
        HaskellSymbolImpl(SYMBOL)
          PsiElement(HaskellTokenType.+)('+')
        HaskellSymbolImpl(SYMBOL)
          PsiElement(HaskellTokenType.+)('+')
    HaskellPstringtokenImpl(PSTRINGTOKEN)
      PsiElement(HaskellTokenType.")('"')
      PsiElement(HaskellTokenType.STRINGTOKEN)(' -> ')
      PsiElement(HaskellTokenType.")('"')
    HaskellQvarsymImpl(QVARSYM)
      HaskellVarsymImpl(VARSYM)
        HaskellSymbolImpl(SYMBOL)
          PsiElement(HaskellTokenType.+)('+')
        HaskellSymbolImpl(SYMBOL)
          PsiElement(HaskellTokenType.+)('+')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('show')
    HaskellQvaridImpl(QVARID)
      HaskellVaridImpl(VARID)
        PsiElement(HaskellTokenType.varidRegexp)('e')
    HaskellQvarsymImpl(QVARSYM)
      HaskellVarsymImpl(VARSYM)
        HaskellSymbolImpl(SYMBOL)
          PsiElement(HaskellTokenType.+)('+')
        HaskellSymbolImpl(SYMBOL)
          PsiElement(HaskellTokenType.+)('+')
    HaskellPstringtokenImpl(PSTRINGTOKEN)
      PsiElement(HaskellTokenType.")('"')
      PsiElement(HaskellTokenType.STRINGTOKEN)(')')
      PsiElement(HaskellTokenType.")('"')
    PsiElement(HaskellTokenType.default)('default')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.()('(')
    HaskellQconidImpl(QCONID)
      PsiElement(HaskellTokenType.conidRegexp)('Integer')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.,)(',')
    HaskellQconidImpl(QCONID)
      PsiElement(HaskellTokenType.conidRegexp)('Double')
    HaskellSpecialImpl(SPECIAL)
      PsiElement(HaskellTokenType.))(')')